packages/blockchain-wallet-v4-frontend/src/components/NetworkWarning/selectors.ts

```diff
@@ -0,0 +1,46 @@
+import { createSelector } from '@reduxjs/toolkit'
+
+import { CoinfigType } from '@core/redux/walletOptions/types'
+import { selectors } from 'data'
+
+const getCoinNetworkIfExistsOnMultiple = (
+  coinfig: CoinfigType,
+  coins: Record<string, { coinfig: CoinfigType }>,
+  evmCompatibleCoinSymbols: string[]
+) => {
+  // Non-native asset
+  if (coinfig?.type?.parentChain) {
+    return coins[coinfig.type.parentChain].coinfig.name
+  }
+
+  // Native asset compatible with EVM
+  if (evmCompatibleCoinSymbols.includes(coinfig.symbol)) {
+    return coinfig.name
+  }
+
+  return null
+}
+
+/**
+ * @returns View model for the warning for the coin if it exists on multiple networks. Otherwise, `null`.
+ */
+export const getCoinNetworkWarning = createSelector(
+  [
+    selectors.core.data.coins.getCoins,
+    selectors.networkConfig.getEvmCompatibleCoins,
+    (_, coin) => coin
+  ],
+  (coins, evmCompatibleCoinSymbols, coin: string) => {
+    const { coinfig } = coins[coin]
+    const maybeNetwork = getCoinNetworkIfExistsOnMultiple(coinfig, coins, evmCompatibleCoinSymbols)
+
+    if (maybeNetwork) {
+      return {
+        network: maybeNetwork,
+        symbol: coinfig.displaySymbol
+      }
+    }
+
+    return null
+  }
+)
```

Comment:
What do you consider different from the current standard in this file?

Personally, what I would like to do with selectors (and other types of functions, e.g. sagas) going forward is instead of putting them all into a single file, split them into separate files under `selectors` folder. This should help with readability, and especially, readability of tests.